context/src/junit/validator.rs

```diff
@@ -149,32 +174,15 @@ pub fn validate(report: &Report) -> JunitReportValidation {
                 ));
             }
 
-            if let Some(timestamp) = test_case
-                .timestamp
-                .or(test_suite.timestamp)
-                .or(report.timestamp)
-            {
-                let now = Utc::now().fixed_offset();
-                let time_since_timestamp = now - timestamp;
-
-                if timestamp > now {
-                    test_case_validation.add_issue(JunitValidationIssue::SubOptimal(
-                        JunitTestCaseValidationIssueSubOptimal::TestCaseFutureTimestamp(timestamp),
-                    ));
-                } else if time_since_timestamp.num_hours() > i64::from(TIMESTAMP_OLD_HOURS) {
-                    test_case_validation.add_issue(JunitValidationIssue::SubOptimal(
-                        JunitTestCaseValidationIssueSubOptimal::TestCaseOldTimestamp(timestamp),
-                    ));
-                } else if time_since_timestamp.num_hours() > i64::from(TIMESTAMP_STALE_HOURS) {
-                    test_case_validation.add_issue(JunitValidationIssue::SubOptimal(
-                        JunitTestCaseValidationIssueSubOptimal::TestCaseStaleTimestamp(timestamp),
-                    ));
-                }
-            } else {
-                test_case_validation.add_issue(JunitValidationIssue::SubOptimal(
-                    JunitTestCaseValidationIssueSubOptimal::TestCaseNoTimestamp,
-                ));
-            }
+            validate_test_case_timestamp(
+                test_case.timestamp,
+                test_suite.timestamp,
+                report.timestamp,
+                test_runner_report,
+                now,
+            )
+            .into_iter()
+            .for_each(|i| test_case_validation.add_issue(JunitValidationIssue::SubOptimal(i)));
```

Comment:
This logic was getting a little large for this function body, so it's been split out